### PR TITLE
Split out aec_chain to use as interface module for chain info

### PR DESCRIPTION
**Pull request by gorillainduction**
Remove external state from aec_chain_state

Redirect calls from aec_conductor to aec_chain

https://www.pivotaltracker.com/n/projects/2124891/stories/154878029

### CODE REVIEW

File: apps/aecore/src/aec_chain_state.erl [hunk 1]
```diff
@@ -811,13 +518,43 @@ find_fork(Hash1, Height1, Hash2, Height2) when Height1 < Height2 ->
         {value, Prev} -> find_fork(Hash1, Height1, Prev, Height2-1)
     end.
 
+%%%-------------------------------------------------------------------
+%%% Persist results
+%%%-------------------------------------------------------------------
+
+persist_chain_state(State) ->
+    aec_db:ensure_transaction(fun() -> persist_state(State)end),
```
Comment (hanssv) [inline apps/aecore/src/aec_chain_state.erl]: missing space before `end`

**Comment by gorillainduction** [comment]
@lucafavatella This is correct. No format of persistent data is changed.

File: apps/aecore/src/aec_conductor.erl [hunk 2]
```diff
@@ -654,24 +416,24 @@ worker_reply(wait_for_keys, Res, State) ->
...
+            update_tx_pool_on_top_change(TopHash, TopBlockHash),
             State1 = State#state{seen_top_block_hash = TopBlockHash},
             State2 = kill_all_workers_with_tag(mining, State1),
             State3 = kill_all_workers_with_tag(create_block_candidate, State2),
             {changed, State3#state{block_candidate = undefined}}
     end.
 
-maybe_publish_top(none,_TopHash,_State) -> ok;
-maybe_publish_top(block_created,_TopHash,_State) ->
+maybe_publish_top(none,_TopHash) -> ok;
```
Comment (hanssv) [inline apps/aecore/src/aec_conductor.erl]: missing space

File: apps/aecore/src/aec_conductor.erl [hunk 3]
```diff
@@ -654,24 +416,24 @@ worker_reply(wait_for_keys, Res, State) ->
...
+maybe_publish_top(block_created,_TopHash) ->
```
Comment (hanssv) [inline apps/aecore/src/aec_conductor.erl]: another one

File: apps/aecore/src/aec_conductor.erl (hunk 2 above)
Comment (gorillainduction) [reply apps/aecore/src/aec_conductor.erl]: No, there isn't ;-)

**Pull request merged**